plugins/hls-hectare-plugin/Ide/Plugin/Hectare.hs

```diff
@@ -0,0 +1,30 @@
+module Ide.Plugin.Hectare (descriptor) where
+
+import           Data.Default
+import           Development.IDE.GHC.Compat
+import           Development.IDE.GHC.Compat.Util
+import qualified ECTA.Plugin                     as GHC
+import           Ide.Plugin.Config
+import           Ide.Types
+
+descriptor :: PluginId -> PluginDescriptor s
+descriptor pluginId = (defaultPluginDescriptor pluginId)
+  { pluginModifyDynflags = staticPlugin
```

Comment:
So _all_ this does is enable the Hectare plugin? I'm a bit confused by this. I would expect that if users want a GHC plugin then they can just add it to their project themselves. It seems a a bit roundabout (and extra work for us) to have HLS plugin for every GHC plugin like that?

Reply:
This is a good question. The Hectare GHC plugin doesn't make programs type check, it does not add a static analysis pass, nor does it rewrite your program to make it faster. It is an IDE plugin, designed to help you write programs faster.

Why is it packaged as a GHC plugin instead of as an HLS plugin then? Well, I guess a GHC plugin is more reusable - anyone can import it in their project, not just HLS users. 

Why repackage it as an HLS plugin? 
 - For convenience, just as we repackage code formatters and linters. 
 - For discoverability, although that is a weaker argument since the plugin is disabled by default.

Reply:
I think the comparison with formatters and linters is interesting:
- For formatters, you get a new way of interacting with the tool, including things like convenient range formatting
- For linters you get the diagnostics integrated with HLS

If we needed this plugin in order for the Hectare diagnostics to work I'd definitely be in favour - it's just that the Hectare developers have been so clever in getting their diagnostics into hole fit suggestions that we don't actually _need_ to do anything if the user turns on the plugin themselves. In no other case is the plugin actually _unnecessary_.

But I do take your argument about it being a pure IDE plugin. There wouldn't be much point enabling it in your cabal file, really!

All of which is to say: I don't know :shrug: so I guess if you're willing to maintain it we can include it!